packages/nvidia_nat_core/src/nat/runtime/user_manager.py

```
@@ -0,0 +1,277 @@
+        Raises:
+            ValueError: If the payload cannot be resolved to a valid user identity.
+        """
+        if isinstance(payload, JwtAuthPayload):
+            raw_token: str = payload.token.get_secret_value()
+            claims: dict[str, typing.Any] = decode_jwt_claims_unverified(raw_token)
+            return UserManager._user_info_from_jwt(claims)
+
+        if isinstance(payload, ApiKeyAuthPayload):
+            token_value: str = payload.token.get_secret_value()
+            if not token_value:
+                raise ValueError("API key token is empty")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)